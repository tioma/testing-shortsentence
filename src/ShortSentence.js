import React from "react";

export default class ShortSentence extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      sentence: "",
      shortSentence: ""
    };
  }

  handleChange = e => {
    const sentence = e.target.value;
    this.setState({ sentence });
  };

  shortSentence = () => {
    const { sentence } = this.state;
    const { length } = this.props;
    let shortSentence = "";

    if (sentence.length > length) {
      shortSentence = sentence.slice(0, length) + "...";
    } else {
      shortSentence = sentence;
    }
    this.setState({ shortSentence });
  };

  render() {
    return (
      <section>
        <h4>Ingresa una frase</h4>
        <input value={this.state.sentence} onChange={this.handleChange} />
        <button onClick={this.shortSentence}>cortar</button>
        <h4 id="frase-cortada">
          la frase cortada es: {this.state.shortSentence}
        </h4>
      </section>
    );--
  }
}
