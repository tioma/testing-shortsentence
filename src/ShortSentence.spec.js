import React from "react";

import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import ShortSentence from "./ShortSentence";

Enzyme.configure({ adapter: new Adapter() });


describe('<ShortSentence />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ShortSentence length={1} />)
  })

  describe('component render', () => {
    it('should match snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    })

    it('should display a button with the label "cortar"', () => {
      expect(wrapper.containsMatchingElement(<button>cortar</button>)).toBe(true);
    });

    it("should have an `input` element", () => {
      expect(wrapper.containsMatchingElement(<input value={""} />)).toBe(true);
    });
  })

  it("should trim a sentence", () => {
    const instance = wrapper.instance();
    const input = wrapper.find('input');
    const button = wrapper.find('button');

    console.log(input.debug());
    console.log(button.length);

    expect(instance.state.sentence).toEqual('');

    const mockEventChange = { target: { value: 'una oración de prueba' }}
    input.simulate('change', mockEventChange);

    expect(instance.state.sentence).toEqual('una oración de prueba')

    button.simulate('click')

    expect(instance.state.shortSentence).toEqual("u...")
  });
})
