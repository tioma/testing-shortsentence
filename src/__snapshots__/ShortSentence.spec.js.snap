// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShortSentence /> component render should match snapshot 1`] = `"<section><h4>Ingresa una frase</h4><input value=\\"\\"/><button>cortar</button><h4 id=\\"frase-cortada\\">la frase cortada es: </h4></section>"`;
